### DISCUSSION TITLE
Quarkus throws InvalidSchemaException when GraphQL has no query

### DISCUSSION THREAD

**Discussion by zbhavyai** [quarkusio/quarkus#25434] [Q&A] [answered]
Hello,

In a REST API, it is optional to have a GET endpoint. However, when I try to do the same with GraphQL, I am getting exception - `InvalidSchemaException invalid schema: "Query" must define one or more fields`. An obvious workaround for this is to create a dummy Query.

I made a small reproducer to replicate the exception I am getting. It is available at [https://github.com/zbhavyai/quarkus-graphql-noquery-reproducer](https://github.com/zbhavyai/quarkus-graphql-noquery-reproducer).

I am using Quarkus versions 2.8.3.Final.

So, I would like to know, is it a GraphQL limitation? Also, can this handled by quarkus/smallrye to avoid making dummy Queries?

Thank you.

[a comment by an automated account is not shown]

**Comment by phillip-kruger** [comment, marked as answer]
This is correct. See https://github.com/smallrye/smallrye-graphql/issues/717

**Reply by zbhavyai** [reply]
Thank you!
